apps/client/src/components/atoms/text/Text.component.tsx

```diff
@@ -0,0 +1,144 @@
+import { ReactElement } from 'react';
+// Interfaces
+import { TextSizeTag, TextTypeTag, TextVariant } from '../../../models';
+// Constants
+import { TextStyles } from '../../../constants';
+// Styles
+import './text.css';
+
+interface TextProps {
+  type: TextTypeTag;
+  size: TextSizeTag;
+  // eslint-disable-next-line react/require-default-props
+  colorClass?: string;
+  children: string;
+}
+
+export function Text({
+  type,
+  size,
+  colorClass,
+  children,
+}: TextProps): ReactElement {
+  const textVariant: TextVariant = `${type}.${size}`;
+
+  interface SubTextProps {
+    size: TextSizeTag;
+    style: React.CSSProperties;
+    className: string;
+    children: string;
+  }
+
+  // === Heading ===
+  const Heading = ({
+    size,
+    style,
+    className,
+    children,
+  }: SubTextProps): React.ReactElement => {
+    if (size === 'Large') {
+      return (
+        <h1 style={style} className={className + ' font-medium'}>
+          {children}
+        </h1>
+      );
+    } else if (size === 'Medium') {
+      return (
+        <h2 style={style} className={className}>
+          {children}
+        </h2>
+      );
+    } else {
+      return (
+        <h3 style={style} className={className}>
+          {children}
+        </h3>
+      );
+    }
+  };
+
+  // === Paragraph ===
+  const Paragraph = ({
+    size,
+    style,
+    className,
+    children,
+  }: SubTextProps): React.ReactElement => {
+    if (size === 'Medium') {
+      return (
+        <p style={style} className={`${className} font-medium text-indent`}>
+          {children}
+        </p>
+      );
+    } else {
+      return (
+        <p style={style} className={className}>
+          {children}
+        </p>
+      );
+    }
+  };
+
+  const textSizeValue = TextStyles.TEXT_MAP.get(textVariant);
+  if (!textSizeValue) {
+    return <span>{children}</span>;
+  } else {
```

Comment:
We don't need the else because returns a value.

Reply:
refactored text component https://github.com/the-ai-team/razor/pull/51/commits/9953c3957419625c23d0ca0cccf993ac3ea6eea9